cmake/modules/Findhidapi.cmake

```diff
@@ -62,28 +61,32 @@ find_library(hidapi_LIBRARY
 )
 mark_as_advanced(hidapi_LIBRARY)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(
-  hidapi
-  DEFAULT_MSG
-  hidapi_LIBRARY
-  hidapi_INCLUDE_DIR
-)
-
 # Version detection
-if (EXISTS "${hidapi_INCLUDE_DIR}/hidapi.h")
-    file(READ "${hidapi_INCLUDE_DIR}/hidapi.h" hidapi_H_CONTENTS)
+if (EXISTS "${hidapi_INCLUDE_DIR}/hidapi/hidapi.h")
+    file(READ "${hidapi_INCLUDE_DIR}/hidapi/hidapi.h" hidapi_H_CONTENTS)
     string(REGEX MATCH "#define HID_API_VERSION_MAJOR ([0-9]+)" _dummy "${hidapi_H_CONTENTS}")
     set(hidapi_VERSION_MAJOR "${CMAKE_MATCH_1}")
     string(REGEX MATCH "#define HID_API_VERSION_MINOR ([0-9]+)" _dummy "${hidapi_H_CONTENTS}")
     set(hidapi_VERSION_MINOR "${CMAKE_MATCH_1}")
     string(REGEX MATCH "#define HID_API_VERSION_PATCH ([0-9]+)" _dummy "${hidapi_H_CONTENTS}")
     set(hidapi_VERSION_PATCH "${CMAKE_MATCH_1}")
     # hidapi_VERSION is only available starting with 0.10.0
-    if (hidapi_VERSION_MAJOR AND hidapi_VERSION_MINOR AND hidapi_VERSION_PATCH)
+    # Simply using if(NOT) does not work because 0 is a valid value, so compare to empty string.
+    if (NOT hidapi_VERSION_MAJOR STREQUAL "" AND
+        NOT hidapi_VERSION_MINOR STREQUAL "" AND
+        NOT hidapi_VERSION_PATCH STREQUAL "")
       set(hidapi_VERSION "${hidapi_VERSION_MAJOR}.${hidapi_VERSION_MINOR}.${hidapi_VERSION_PATCH}")
     endif()
-endif ()
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(
+  hidapi
+  DEFAULT_MSG
+  hidapi_LIBRARY
+  hidapi_INCLUDE_DIR
+  hidapi_VERSION
```

Comment:
This makes the hidapi_VERSION version required to "find" the package which is an issue because the code above does not work for versions < 0.10.0. We can make it work by using the version from the PC file or just stick with the original solution.

Reply:
Please give a hint what you prefer and I will prepare a PR.

Reply:
This is intentional. We require 0.10.1, so if there is no hidapi_VERSION it is too old anyway.